webapp/src/index.tsx

```diff
@@ -11,20 +12,95 @@ import EditorMenu from './components/editor_menu';
 import IconReactForMe from './components/assets/icon_react_for_me';
+import IconAI from './components/assets/icon_ai';
+import RHS from './components/rhs/rhs';
 import Config from './components/config/config';
-import {doReaction, doSummarize, doTranscribe} from './client';
+import {doReaction, doSummarize, doTranscribe, getAIDirectChannel} from './client';
+import {setOpenRHSAction} from './redux_actions';
 import {BotUsername} from './constants';
 import PostEventListener from './websocket';
 
 type WebappStore = Store<GlobalState, Action<Record<string, unknown>>>
 
 const StreamingPostWebsocketEvent = 'custom_mattermost-ai_postupdate';
 
+const IconAIContainer = styled.span`
+    filter: invert(1);
+    border-radius: 50%;
+    width: 28px;
+    height: 28px;
+    display: inline-flex;
+    align-items: center;
+    justify-items: center;
+    margin-right: 5px;
+    background-color: var(--center-channel-bg);
+`;
+
+const RHSTitleContainer = styled.span`
+    display: flex;
+    align-items: center;
+`;
+
+const RHSTitle = () => {
+    return (
+        <RHSTitleContainer>
+            <IconAIContainer className='icon'>
+                <IconAI/>
+            </IconAIContainer>
+            {'AI Assistant'}
+        </RHSTitleContainer>
+    );
+};
+
 export default class Plugin {
     postEventListener: PostEventListener = new PostEventListener();
 
     // eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/no-empty-function
     public async initialize(registry: any, store: WebappStore) {
+        let rhs: any = null;
+        if ((window as any).Components.CreatePost) {
+            rhs = registry.registerRightHandSidebarComponent(RHS, RHSTitle);
+            setOpenRHSAction(rhs.showRHSPlugin);
+
+            registry.registerReducer((state = {}, action: any) => {
+                switch (action.type) {
+                case 'SET_AI_BOT_CHANNEL':
+                    return {
+                        ...state,
+                        botChannelId: action.botChannelId,
+                    };
+                case 'SELECT_AI_POST':
+                    return {
+                        ...state,
+                        selectedPostId: action.postId,
+                    };
+                default:
+                    return state;
+                }
+            });
+        }
+
+        let currentUserId = store.getState().entities.users.currentUserId;
+        if (currentUserId) {
+            getAIDirectChannel(currentUserId).then((botChannelId) => {
+                store.dispatch({type: 'SET_AI_BOT_CHANNEL', botChannelId} as any);
+            });
+        }
+
+        store.subscribe(() => {
+            const state = store.getState();
+            if (state && state.entities.users.currentUserId !== currentUserId) {
+                currentUserId = state.entities.users.currentUserId;
+                if (currentUserId) {
+                    getAIDirectChannel(currentUserId).then((botChannelId) => {
```

Comment:
Not sure about this janky client side solution to getting the channel name so we can get the posts later. Maybe it would be better if we created an API endpoint on the plugin side so we can just ask for what to display. Then we can fetch the posts easily server side.

Also allows for possible integrations with external systems.

Reply:
Good point, let me explore that.

Reply:
This can solve multiple problems here, like the fact that we have to relay on CRT or not CRT enable to be able to get a certain amount of root posts, and after that, that can lead to more complex pagination. But having something like "AI Conversations" endpoint would help. I don't see the need of doing it right away, because this PR already have enough complexity around UI, but I would love to have that new endpoint in another PR later.